src/utils/fatal.rs

```diff
@@ -322,9 +322,17 @@ pub fn write_file(path: &Path, contents: &[u8]) -> Result<(), Failed> {
 ///
 /// Creates the target directory with `create_dir_all`.  Errors out if
 /// anything goes wrong.
-pub fn copy_dir_all(source: &Path, target: &Path) -> Result<(), Failed> {
+///
+/// If the source director does not exist, does nothing.
```

Comment:
s/director/directory/

Other than that, I find this a bit confusing. It has been explicitly renamed to say it operates on an +_existing_+ directory, yet checks if it exists and succeeds without doing anything if it doesn't exist. That violates my expectations based on the name.

Reply:
The intention was to have a function that doesn’t error if something doesn’t exist. I adjusted the doc string slightly.